protocols/identify/src/handler.rs

```diff
@@ -315,10 +315,6 @@ impl ConnectionHandler for Handler {
     }
 
     fn connection_keep_alive(&self) -> KeepAlive {
-        if !self.active_streams.is_empty() {
```

Comment:
Great! Now we just need to do something similar for **all** other protocols, i.e. everything under `protocols/`!

Let me know if you've got any questions :)

Reply:
Kinda confused about this one. We need to go through all implementations of ConnectionHandler `connection_keep_alive` function and remove any duplicate stream tracking code?

Reply:
> We need to go through all implementations of ConnectionHandler `connection_keep_alive` function and remove any duplicate stream tracking code?

At some point, yes! The end goal of this effort is to make `connection_keep_alive` optional to implement. See https://github.com/libp2p/rust-libp2p/issues/4306.

I think for the most part this should be fairly trivial. If you run into any more complicated ones, feel free to raise a comment. We don't need to strictly remove it within this PR as nothing fails if it is still there. It is just redundant code now because we are doing all the stream-tracking within `Connection`.

We can definitely postpone some of this into follow-up issues but it would be great to get at least all the easy ones like `libp2p-identify` done!

Reply:
remove stream tracking code c931771